src/openforms/config/migrations/0059_convert_button_design_tokens.py

```diff
@@ -0,0 +1,142 @@
+    "utrecht.button.secondary-action.danger.hover.background-color": "of.button.danger.hover.bg",
+    "utrecht.button.secondary-action.danger.hover.border-color": "of.button.danger.hover.color-border",
+    "utrecht.button.secondary-action.danger.active.background-color": "of.button.danger.active.bg ",
+    "utrecht.button.secondary-action.danger.active.border-color": "of.button.danger.active.color-border ",
+    "utrecht.button.secondary-action.danger.active.color": "of.button.danger.active.fg",
+    "utrecht.button.secondary-action.danger.focus.border-color": "of.button.danger.focus.color-border ",
+    "utrecht.button.subtle.danger.background-color": "of.button.danger.bg",
+    "utrecht.button.subtle.danger.border-color": "of.button.danger.color-border",
+    "utrecht.button.subtle.danger.color": "of.color.danger",
+    "utrecht.button.subtle.danger.active.background-color": "of.color.danger",
+    "utrecht.button.subtle.danger.active.color": "of.color.bg",
+}
+
+
+class TokenValue(TypedDict):
+    value: str
+
+
+TokenContainer: TypeAlias = dict[str, "TokenValue | TokenContainer"]
+
+
+unset = object()
+
+
+def apply_mapping(design_tokens: TokenContainer) -> TokenContainer:
+    result = deepcopy(design_tokens)
+
+    tokens_to_unset = set()
+
+    for new, old in MAPPING.items():
+        old_value = glom(design_tokens, old, default=unset)
+        if old_value is unset:
+            continue
+
+        existing_value = glom(result, new, default=unset)
+        if existing_value is not unset:
+            tokens_to_unset.add(old)
+            continue
+
+        assign(result, new, old_value, missing=dict)
+        tokens_to_unset.add(old)
+
+    for token in tokens_to_unset:
+        # don't delete utility tokens!
+        if not token.startswith("of.button."):
+            continue
+        delete(result, token)
+
+    return remove_empty_design_tokens(result)
+
+
+def update_design_token_values(apps, _):
+    GlobalConfiguration = apps.get_model("config", "GlobalConfiguration")
+    config = GlobalConfiguration.objects.first()
+    if config is None:
+        return
+
+    updated = apply_mapping(config.design_token_values)
+    if updated != config.design_token_values:
+        config.design_token_values = updated
+        config.save(update_fields=["design_token_values"])
+
+
+def remove_empty_design_tokens(obj: dict) -> dict:
+    if "value" in obj:
+        return obj
+
+    result = {}
+    for key, value in obj.items():
+        if not isinstance(value, dict):
+            continue
+        updated_value = remove_empty_design_tokens(value)
+        # empty object -> remove it by not including it anymore
+        if not updated_value:
+            continue
+
+        result[key] = updated_value
+
+    return result
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("config", "0058_auto_20231026_1525"),
+    ]
+
+    operations = [
+        # reversing this is not possible - make sure you have backups!
```

Comment:
Maybe a good idea to mention this in the docs too? I guess that if they have specific design tokens and after the migration the style is off, it is nice for the form designers to have a copy of the old tokens rather than just in the database backup? :thinking: